plugins/ingest-attachment/src/main/java/org/elasticsearch/ingest/attachment/AttachmentProcessor.java

```diff
@@ -0,0 +1,202 @@
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.elasticsearch.ingest.core.ConfigurationUtils.newConfigurationException;
+import static org.elasticsearch.ingest.core.ConfigurationUtils.readIntProperty;
+import static org.elasticsearch.ingest.core.ConfigurationUtils.readOptionalList;
+import static org.elasticsearch.ingest.core.ConfigurationUtils.readStringProperty;
+
+public final class AttachmentProcessor extends AbstractProcessor {
+
+    public static final String TYPE = "attachment";
+
+    private static final int NUMBER_OF_CHARS_INDEXED = Integer.MAX_VALUE;
+
+    private final String sourceField;
+    private final String targetField;
+    private final Set<Field> fields;
+    private final int indexedChars;
+
+    AttachmentProcessor(String tag, String sourceField, String targetField, Set<Field> fields,
+                        int indexedChars) throws IOException {
+        super(tag);
+        this.sourceField = sourceField;
+        this.targetField = targetField;
+        this.fields = fields;
+        this.indexedChars = indexedChars;
+    }
+
+    @Override
+    public void execute(IngestDocument ingestDocument) {
+        String base64Input = ingestDocument.getFieldValue(sourceField, String.class);
+
+        Metadata metadata = new Metadata();
+        try {
+            byte[] decodedContent = Base64.decode(base64Input.getBytes(UTF_8));
+            String parsedContent = TikaImpl.parse(decodedContent, metadata, indexedChars);
+
+            if (fields.contains(Field.CONTENT)) {
+                // somehow tika seems to append a newline at the end automatically, lets remove that again
+                ingestDocument.setFieldValue(targetField + "." + Field.CONTENT.toLowerCase(), parsedContent.trim());
+            }
+
+            if (fields.contains(Field.LANGUAGE)) {
+                LanguageIdentifier identifier = new LanguageIdentifier(parsedContent);
+                String language = identifier.getLanguage();
+                ingestDocument.setFieldValue(targetField + "." + Field.LANGUAGE.toLowerCase(), language);
+            }
+
+            if (fields.contains(Field.DATE)) {
+                String createdDate = metadata.get(TikaCoreProperties.CREATED);
+                if (createdDate != null) {
+                    ingestDocument.setFieldValue(targetField + "." + Field.DATE.toLowerCase(), createdDate);
+                }
+            }
+
+            if (fields.contains(Field.TITLE)) {
+                String title = metadata.get(TikaCoreProperties.TITLE);
+                if (Strings.hasLength(title)) {
+                    ingestDocument.setFieldValue(targetField + "." + Field.TITLE.toLowerCase(), title);
+                }
+            }
+
+            if (fields.contains(Field.AUTHOR)) {
+                String author = metadata.get("Author");
```

Comment:
I know. Got the same problem with mapper attachment plugin. 
That why we added `@SuppressWarnings("deprecation") // https://github.com/elastic/elasticsearch/issues/15843`.

I think we should use the same here. Otherwise we won't be able to know if at some point getting `Author` from Metadata is supposed to work or not.
If Tika definitely removes it, we won't be able to compile the code anymore when upgrading the lib, so we'll be aware of it and we would be able to take decisions accordingly.